niworkflows/interfaces/cifti.py

```diff
@@ -68,196 +84,42 @@ class GenerateCifti(SimpleInterface):
     """
     Generate CIFTI image from BOLD file in target spaces.
 
-    Currently supported
-    * target surfaces: fsaverage5, fsaverage6
-    * target volumes: OASIS-TRT-20_DKT31 labels in MNI152NLin2009cAsym
+    Currently supports ``fsLR``, ``fsaverage5``, or ``fsaverage6`` for template surfaces and
+    ``MNI152NLin6Asym`` or ``MNI152NLin2009cAsym`` as template volumes.
 
     """
 
     input_spec = _GenerateCiftiInputSpec
     output_spec = _GenerateCiftiOutputSpec
 
     def _run_interface(self, runtime):
-        self._results["variant_key"], self._results["variant"] = self._define_variant()
-        annotation_files, label_file = self._fetch_data()
-        self._results["out_file"] = self._create_cifti_image(
+        annotation_files, label_file = _get_cifti_data(
+            self.inputs.surface_target,
+            self.inputs.volume_target,
+            self.inputs.subjects_dir,
+            self.inputs.density
+        )
+        self._results['variant_key'], self._results['variant'] = _get_cifti_variant(
```

Comment:
feels like `variant_key` and `variant` meanings are swapped. The `_key` suffix makes me think of a string not a file.